giotto/homology/grids.py

```diff
@@ -0,0 +1,182 @@
+                 infinity_values=None, n_jobs=None):
+        self.homology_dimensions = homology_dimensions
+        self.coeff = coeff
+        self.infinity_values = infinity_values
+        self.n_jobs = n_jobs
+
+    def _gudhi_diagram(self, X):
+        cubical_complex = CubicalComplex(
+            dimensions=X.shape,
+            top_dimensional_cells=X.flatten(order="F"))
+        Xds = cubical_complex.persistence(homology_coeff_field=self.coeff,
+                                          min_persistence=0)
+        Xds = {dim: np.array([Xds[i][1] for i in range(len(Xds))
+                              if Xds[i][0] == dim]).reshape((-1, 2))
+               for dim in self.homology_dimensions}
+        Xds = {dim: np.hstack([Xds[dim], dim * np.ones((Xds[dim].shape[0], 1),
+                                                       dtype=Xds[dim].dtype)])
+               for dim in self._homology_dimensions}
+        return Xds
+
+    def fit(self, X, y=None):
+        """Do nothing and return the estimator unchanged.
+
+        This method is there to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_pixels_1, ..., n_pixels_d)
+            Input data. Array of d-dimensional images.
+
+        y : None
+            There is no need of a target in a transformer, yet the pipeline API
+            requires this parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        if self.infinity_values is None:
+            self.infinity_values_ = np.max(X)
+        else:
+            self.infinity_values_ = self.infinity_values
+
+        self._homology_dimensions = sorted(self.homology_dimensions)
+
+        validate_params({**self.get_params(),
+                         'infinity_values_': self.infinity_values_,
+                         '_homology_dimensions': self._homology_dimensions},
+                        self._hyperparameters)
+        check_array(X, allow_nd=True)
+
+        self._max_homology_dimension = self._homology_dimensions[-1]
+        return self
+
+    def transform(self, X, y=None):
+        """Compute, for each image in `X`, the relevant persistence diagram
+        as an array of triples [b, d, q]. Each triple represents a persistent
+        topological feature in dimension q (belonging to `homology_dimensions`)
+        which is born at b and dies at d. Only triples in which b < d are
+        meaningful. Triples in which b and d are equal ("diagonal elements")
+        may be artificially introduced during the computation for padding
+        purposes, since the number of non-trivial persistent topological
+        features is typically not constant across samples. They carry no
+        information and hence should be effectively ignored by any further
+        computation. (b, d) -- one per persistent topological hole -- where b
```

Comment:
I would write this as "For each persistent topological hole, b is the scale ..."

Reply:
@ulupo @lewtun I think this description is also all over the library. Maybe we should be consistent and improve it everywhere at the same time?

Reply:
This additional line was a mistake on my side.